### PR TITLE
`azurerm_spring_cloud_gateway` - use pandora SDK

### PR DESCRIPTION
**Pull request by ms-henglu**
image

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 1]
```diff
@@ -4,697 +4,787 @@
...
 
-	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/migration"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/parse"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/validate"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/appplatform/2023-09-01-preview/appplatform"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:14]: We shall stick on using `pluginsdk` instead.

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 2]
```diff
@@ -4,697 +4,787 @@
...
-		return fmt.Errorf("checking for presence of existing Spring Cloud Service %q (Resource Group %q): %+v", springId.SpringName, springId.ResourceGroup, err)
-	}
-	if service.Sku == nil || service.Sku.Name == nil || service.Sku.Tier == nil {
-		return fmt.Errorf("invalid `sku` for Spring Cloud Service %q (Resource Group %q)", springId.SpringName, springId.ResourceGroup)
-	}
+			service, err := client.ServicesGet(ctx, *springId)
+			if err != nil {
+				return fmt.Errorf("checking for presence of existing %s: %+v", springId, err)
+			}
+			if service.Model.Sku == nil || service.Model.Sku.Name == nil || service.Model.Sku.Tier == nil {
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:392]: We shall also nil check `service.Model`

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 3]
```diff
@@ -4,697 +4,787 @@
...
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			state := SpringCloudGatewayModel{
+				Name:                          id.GatewayName,
+				SpringCloudServiceId:          springId.ID(),
+				SensitiveEnvironmentVariables: model.SensitiveEnvironmentVariables,
+			}
+
+			if props := resp.Model.Properties; props != nil {
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:545]: We shall also nil check `resp.Model`.

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

**Comment by ms-henglu** [comment]
https://github.com/hashicorp/terraform-provider-azurerm/pull/24116

**Pull request closed without merging**